\nonstopmode{}
\documentclass[letterpaper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `AFLPsim'}}
\par\bigskip{\large \today}
\end{center}
\begin{description}
\raggedright{}
\inputencoding{latin1}
\item[Type]\AsIs{Package}
\item[Title]\AsIs{Hybrid simulation and genome scan for dominant markers}
\item[Version]\AsIs{0.1-1}
\item[Encoding]\AsIs{latin1}
\item[Date]\AsIs{2013-05-05}
\item[Author]\AsIs{Francisco  Balao and Juan Luis Garcia-Castaño}
\item[Maintainer]\AsIs{Francisco Balao }\email{fbalao@us.es}\AsIs{}
\item[Depends]\AsIs{introgress}
\item[Description]\AsIs{This package is developed in the Plant Reproductive Biology Lab (RNM-214) - University of Seville. It contains hybrid simulation functions for dominant genetic data. It also provides several genome scan methods.}
\item[License]\AsIs{GPL (>= 2)}
\item[URL]\AsIs{}\url{http://www.r-project.org,}\AsIs{
}\url{http://personal.us.es/fbalao/fbalaoenglish.html}\AsIs{}
\item[LazyData]\AsIs{true}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{AFLPsim-package}{Hybrid simulation and genome scan for dominant markers}{AFLPsim.Rdash.package}
\keyword{package}{AFLPsim-package}
%
\begin{Description}\relax
This package is developed in the Plant Reproductive Biology Lab (RNM-214) - University of Seville. It contains hybrid simulation functions for dominant genetic data. It also provides several genome scan methods. 

More information on AFLPsim can be found at \url{https://github.com/fbalao/AFLPsim}.

To cite AFLPsim, please use citation("AFLPsim")
\end{Description}
%
\begin{Details}\relax

\Tabular{ll}{
Package: & AFLPsim\\{}
Type: & Package\\{}
Version: & 0.1-1\\{}
Date: & 2013-05-21\\{}
License: & GPL (>= 2)\\{}
}
\end{Details}
%
\begin{Author}\relax
Francisco Balao; Juan Luis Garcia CastaÃ±o
\end{Author}
\inputencoding{utf8}
\HeaderA{bayescan}{Identifying candidate loci under natural selection with External Application}{bayescan}
\keyword{outlier}{bayescan}
\keyword{hybridization}{bayescan}
\keyword{genome scan}{bayescan}
%
\begin{Description}\relax
These functions call Bayescan program from R to identifying candidate loci under natural selection from genetic data.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
bayescan(mat, filename, nbp = 20, pilot = 5000, burn = 50000, exec=NULL)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{mat}] 
A matrix with genotypic data to test

\item[\code{filename}] 
a character string giving the name of the output file

\item[\code{nbp}] 
Number of pilot runs, default is 2 

\item[\code{pilot}] 
Length of pilot runs, default is 50

\item[\code{burn}] 
Burnin length, default is 5000

\item[\code{exec}] 
a character string giving the path to bayescan, default it tries to guess it depending on the operating system (see details).

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
bayescan tries to guess the name of the executable program depending on the operating system. Specifically, the followings are used: â/home/BayeScan2.1/binaries/BayeScan2.1\_linux32bits"â under Linux, or âC:/Program Files/BayeScan2.1/binaries/BayeScan2.1\_win32bits\_cmd\_line.exeâ under Windows. For Mac is under develop
\end{Details}
%
\begin{Value}
A list with the following components:
\begin{ldescription}
\item[\code{resultfst}] XXXXXXX
\item[\code{outliers}] a vector with outliers
\item[\code{TP}] XXXXXXXXXX
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Foll, M., and O. Gaggiotti. 2008. A genome-scan method to identify selected loci appropriate for both dominant and codominant markers: a Bayesian perspective. \emph{Genetics} \bold{180}: 977â93. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{gscan}{gscan}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=100, Na=30, Nb=30, Nf1=30, type="selection", S=4,Nsel=25, hybrid="F1")

outliers<-gscan(hybrids, type="F1", method="balao")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{demosimhybrid}{Demographic model of introgressive hybridization }{demosimhybrid}
\keyword{hybridization}{demosimhybrid}
\keyword{simulation}{demosimhybrid}
%
\begin{Description}\relax
This model simulates the proportions of parentals,
F1, F2, Fx and backcross (on both sides) individuals for each generation and takes into account the initial frequencies of parentals, the assortative mating among taxa as well as fitness differences.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
demosimhybrid(x,M,F)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
A vector indicating the initial abundances in the population.The vector should sum 1. The order of abundances is: ParentalA, ParentalB, F1, BxA, BxB and Fx

\item[\code{M}] 
Matrix assortative mating.The size is 6x6 following the same order tan vector x. By default random mating (all = 1)

\item[\code{F}] 
A vector indicating the diferent fecundities of the parentals and hybrids. The vector size is 6 following the same order tan vector x. By default equal fecundities (all = 1)

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
This function simulate the model of introgresive hibridization of Epifanio and Philipp (2000)
\end{Details}
%
\begin{Value}
Object \code{demosimhybrid}. An matrix of abundances in each generation
\end{Value}
%
\begin{Author}\relax
Francisco Balao \email{fbalao@us.es};
Marcial Escudero
\end{Author}
%
\begin{References}\relax
Epifanio, J. and D. Philipp. 2000. Simulating the extinction of parental lineages from introgressive hybridization: the effects of fitness, initial proportions of parental taxa, and mate choice. \emph{Reviews in Fish Biology and Fisheries} \bold{10}: 339â354. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{hybridsim}{hybridsim}}

\code{\LinkA{plot.demosimhybrid}{plot.demosimhybrid}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}

#Example 1. Simulation under parental proportions, similar fecundities and random mating 
inivalues<-c(0.5,0.5,0,0,0,0)
epi0.5<-demosimhybrid(inivalues)
epi0.5
plot(epi0.5)

#Example 2. Simulation under higher frecuency of parental B, and higher fecundy of parental A and random mating
inivalues2<-c(0.25,0.75,0,0,0,0)
fecundities<-c(1,0.5,0.5,0.5,0.5,0.5)
epi0.75<-demosimhybrid(x=inivalues, F=fecundities)
epi0.75
plot(epi0.75)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{gscan}{Genome scan for hybrids}{gscan}
\keyword{outlier}{gscan}
\keyword{hybridization}{gscan}
\keyword{genome scan}{gscan}
%
\begin{Description}\relax
This function fits genomic scan to dominant genotypic data using the method described by Gagnaire et al (2009) and the new method by Balao et al (2013). Significance testing for outlier loci is included.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
gscan(mat, type=c("F1","BxA","BxB"), method=c("balao","gagnaire"))

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{mat}] 
A hybridsim object produced by hybridsim or hybridize

\item[\code{type}] 
type of hybrid classes; either "F1", "BxA" or "BxB"

\item[\code{method}] 
method to test significance of outlier loci; either "gagnaire" or "balao". See Details.

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
These genome scan methods calculate the null distribution of frequencies under a neutral model. A binomial test is used to outlier significance. 
\end{Details}
%
\begin{Value}
A list with the following components:
\begin{ldescription}
\item[\code{P-values }] a matrix with p-values after False Discovery correction for each loci
\item[\code{Outlier }] a vector with outliers
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Balao, F., Casimiro-Soriguer, R., Garcia-CastaÃ±o, J.L., Terrab, A., Talavera, S. 2013. Big thistle eats the little thistle: Non-neutral unidirectional introgression endangers the conservation of \emph{Onopordum hinojense}. \emph{Molecular Ecology}, \emph{in preparation}.

Benjamini, Y., and Y. Hochberg. 1995. Controlling the false discovery rate: a practical and powerful approach to multiple testing. \emph{Journal of the Royal Statistical Society. Series B} \bold{57}: 289â300.

Gagnaire, P.A., V. Albert, B. JÃ³nsson, L. Bernatchez. 2009. Natural selection influences AFLP intraspecific genetic variability and introgression patterns in Atlantic eels. \emph{Molecular Ecology} \bold{18}: 1678â1691.

\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{hybridsim}{hybridsim}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=100, Na=30, Nb=30, Nf1=30, type="selection", S=5,Nsel=25, hybrid="F1")

outliers<-gscan(hybrids, type="F1", method="balao")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{hybridindex}{Estimate Hybrid Index por AFLPsim objects}{hybridindex}
\keyword{hybridization}{hybridindex}
%
\begin{Description}\relax
This function finds maximum likelihood estimates of hybrid index as described by Buerkle (2005) using the packages "introgress"
\end{Description}
%
\begin{Usage}
\begin{verbatim}
hybridindex(hybriddata)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{hybriddata}] 
A AFLPsim object with Parentals and hybrids profiles

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Returns a hibrid index estimate and 95\% confidence interval of hybrid index.

See Buerkle (2005) for additional details.
\end{Details}
%
\begin{Value}
A data frame with point estimates of hybrid index and upper and lower limits of 95\% confidence intervals (interval of hybrid index that falls within two support units of the ML estimate):

\begin{ldescription}
\item[\code{lower}]   
Lower limit of 95\% confidence interval.

\item[\code{h}] 
Maximum-likelihood estimate of hybrid index.

\item[\code{upper}] 
Upper limit of 95\% confidence interval.
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
F. Balao;
Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Balao
Buerkle 2005
https://github.com/fbalao/AFLPsim
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{est.h}{est.h}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=300, Na=100, Nb=100, Nf1=100, type="neutral", hybrid="F1")

hest<-hybridindex(hybrids)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{hybridize}{AFLP Simulator wih selection from parental data}{hybridize}
\keyword{hybridization}{hybridize}
\keyword{simulation}{hybridize}
%
\begin{Description}\relax
Simulate AFLP profiles (or other dominant markers) several hybrid classes (F1 and backcrosses on both parentals) from two parental populations. In addition, selection on several markers could be simulated in hybrid progeny.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
hybridize(pa,pb,Nf1,Nbxa, Nbxb, Nf2, type,hybrid,Nsel,S)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{pa}] 
AFLP profiles of parental 1 population. A data.frame or matrix

\item[\code{pb}] 
AFLP profiles of parental 2 population. A data.frame or matrix

\item[\code{Nf1}] 
Number of AFLP profiles to F1 population

\item[\code{Nbxa}] 
Number of AFLP profiles to BxA population

\item[\code{Nbxb}] 
Number of AFLP profiles to BxB population

\item[\code{Nf2}] 
Number of AFLP profiles to F2 population

\item[\code{type}] 
Type of simulation: "neutral" or "selection"

\item[\code{hybrid}] 
Hybrid classes to simulate. By default "all". Also "F1", "BxA", "BxB" and "F2".


\item[\code{Nsel}] 
Number of loci under selection. Must be less than Nmarker

\item[\code{S}] 
Selection coefficient (see Details)

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Simulate F1, F2, Backcrosses to parental A (BxA) and Backcrosses to parental b (BxB) with and without selection. Parental profiles must be included. The selection coefficient S is a integer value from -10 to 10 (negative and positive selection). 
\end{Details}
%
\begin{Value}
Object \code{hybridsim} with:
\begin{ldescription}
\item[\code{PA }] AFLP profile parental population A
\item[\code{PB }] AFLP profile parental population A
\item[\code{F1}] AFLP profile F1 hybrid population
\item[\code{F2}] AFLP profile F2 hybrid population
\item[\code{BxA}] AFLP profile Backcrosses to parental population A
\item[\code{BxB}] AFLP profile Backcrosses to parental population B
\item[\code{Nsel}] Loci under selection. NA under neutral selection
\item[\code{S}] Selection Coefficient
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Wright, S. 1931. Evolution in Mendelian Populations. \emph{Genetics} \bold{16}: 97â159. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{hybridsim}{hybridsim}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
#AFLP profile Species A
SpeciesA<-read.table(system.file("files/SpeciesA.txt",package="AFLPsim"),header=TRUE, row.names=1)


#AFLP profile Species B
SpeciesB<-read.table(system.file("files/SpeciesB.txt",package="AFLPsim"),header=TRUE, row.names=1)

F1hybrids<-hybridize(pa=SpeciesA,pb=SpeciesB, Nf1=30, type="neutral", hybrid="F1")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{hybridsim}{AFLP Simulator wih selection}{hybridsim}
\keyword{hybridization}{hybridsim}
\keyword{simulation}{hybridsim}
%
\begin{Description}\relax
Simulate AFLP profiles (or other dominat markers) of two parental populations and several hybrid classes (F1 and backcrosses on both parentals). In addition, selection on several markers could be simulated in hybrid progeny.  
\end{Description}
%
\begin{Usage}
\begin{verbatim}
hybridsim(Nmarker, Na, Nb, Nf1, Nbxa = Nf1, Nbxb = Nf1, Nf2 = Nf1, type = "selection", hybrid = "all", Nsel = Nmarker * 0.1, S = 0, apa = 0.5, apb = 0.5)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{Nmarker}] 
The number of AFLP loci to simulate

\item[\code{Na}] 
Number of AFLP profiles to parental 1 population

\item[\code{Nb}] 
Number of AFLP profiles to parental 2 population

\item[\code{Nf1}] 
Number of AFLP profiles to F1 population

\item[\code{Nbxa}] 
Number of AFLP profiles to BxA population

\item[\code{Nbxb}] 
Number of AFLP profiles to BxB population

\item[\code{Nf2}] 
Number of AFLP profiles to F2 population

\item[\code{type}] 
Type of simulation: "neutral" or "selection"

\item[\code{hybrid}] 
Hybrid classes to simulate. By default "all". Also "F1", "BxA", "BxB" and "F2".


\item[\code{apa}] 
alpha value in beta distribution for parental 1 population

\item[\code{apb}] 
alpha value in beta distribution for parental 2 population

\item[\code{Nsel}] 
Number of loci under selection. Must be less than Nmarker

\item[\code{S}] 
Selection coefficient (see Details)


\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Simulate dominant markers (AFLP, RFLP...). Parental allele frequencies are calculated following a beta distribution (Wright 1931).
F1, F2, Backcrosses to parental A (BxA) and Backcrosses to parental b (BxB) can be simulated with and without selection. The selection coefficient S is a integer value from -1 to 1 (negative and positive selection).  
\end{Details}
%
\begin{Value}
Object \code{hybridsim}  with:
\begin{ldescription}
\item[\code{PA }] AFLP profile parental population A
\item[\code{PB }] AFLP profile parental population A
\item[\code{F1}] AFLP profile F1 hybrid population
\item[\code{F2}] AFLP profile F2 hybrid population
\item[\code{BxA}] AFLP profile Backcrosses to parental population A
\item[\code{BxB}] AFLP profile Backcrosses to parental population B
\item[\code{Nsel}] Loci under selection. NA under neutral selection
\item[\code{S}] Selection Coefficient
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
F. Balao;
Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Wright, S. 1931. Evolution in Mendelian Populations. \emph{Genetics} \bold{16}: 97â159. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{hybridize}{hybridize}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=100, Na=30, Nb=30, Nf1=30, type="selection", Nsel=25, hybrid="F1")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plot.demosimhybrid}{Plotting demosimhybrid objects}{plot.demosimhybrid}
\keyword{hybridization}{plot.demosimhybrid}
\keyword{simulation}{plot.demosimhybrid}
%
\begin{Description}\relax
\code{\LinkA{demosimhybrid}{demosimhybrid}} object can be plotted using the function \code{demosim.hybrid.plot}, which is also used as the dedicated plot method. These functions plot the frequency of parentals and hybrid classes on each generation.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plot.demosimhybrid(x, col = c(2, 3, 4, "orange", "orchid", 7))
## S3 method for class 'demosimhybrid'
plot(x, col = c(2, 3, 4, "orange", "orchid", 7))
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
A \code{\LinkA{demosimhybrid}{demosimhybrid}} object 

\item[\code{col}] 
a color vector to be used to represent hibrid classes.

\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
This function simulate the model of introgresive hibridization of Epifanio and Philipp (2000)
\end{Details}
%
\begin{Value}
Object 'demosimhybrid'. An matrix of abundances in each generation
\end{Value}
%
\begin{Author}\relax
Francisco Balao \email{fbalao@us.es};
Marcial Escudero
\end{Author}
%
\begin{References}\relax
Epifanio, J. and D. Philipp. 2000. Simulating the extinction of parental lineages from introgressive hybridization: the effects of fitness, initial proportions of parental taxa, and mate choice. \emph{Reviews in Fish Biology and Fisheries} \bold{10}: 339â354. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{demosimhybrid}{demosimhybrid}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}

#Example 1. Simulation under parental proportions, similar fecundities and random mating 
inivalues<-c(0.5,0.5,0,0,0,0)
epi0.5<-demosimhybrid(inivalues)
epi0.5
plot(epi0.5)

#Example 2. Simulation under higher frecuency of parental B, and higher fecundy of parental A and random mating
inivalues2<-c(0.25,0.75,0,0,0,0)
fecundities<-c(1,0.5,0.5,0.5,0.5,0.5)
epi0.75<-demosimhybrid(x=inivalues, F=fecundities)
epi0.75
plot(epi0.75)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plot.hybridsim}{Plotting hybridsim objects}{plot.hybridsim}
\keyword{hybridization}{plot.hybridsim}
\keyword{simulation}{plot.hybridsim}
%
\begin{Description}\relax
\code{\LinkA{hybridsim}{hybridsim}} object can be plotted using the function \code{hybridsim.plot}, which is also used as the dedicated plot method. These functions relie on image to represent hybrid markers frequency on a neutral expectation sheet.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plot.hybridsim(x, hybrid = c("F1", "BxA", "BxB"), col = "lightgreen", shade = 0.8, markers = x$SelMarkers) 
## S3 method for class 'hybridsim'
plot(x,hybrid=c("F1","BxA","BxB"),col="lightgreen",shade=0.8, markers= x$SelMarkers)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{x}] 
A \code{hybridsim} object

\item[\code{hybrid}] 
Hybrid classe to simulate, "F1", "BxA" or "BxB".

\item[\code{col}] 
A specification for the default plotting color

\item[\code{shade}] 
A specification for the default alpha value

\item[\code{markers}] 
numeric vectors with markers to plot


\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Simulate dominant markers (AFLP, RFLP...). Parental allele frequencies are calculated following a beta distribution (Wright 1931).
F1, F2, Backcrosses to parental A (BxA) and Backcrosses to parental b (BxB) can be simulated with and without selection. The selection coefficient S is a integer value from -1 to 1 (negative and positive selection).  
\end{Details}
%
\begin{Value}
Object \code{hybridsim}  with:
\begin{ldescription}
\item[\code{PA }] AFLP profile parental population A
\item[\code{PB }] AFLP profile parental population A
\item[\code{F1}] AFLP profile F1 hybrid population
\item[\code{F2}] AFLP profile F2 hybrid population
\item[\code{BxA}] AFLP profile Backcrosses to parental population A
\item[\code{BxB}] AFLP profile Backcrosses to parental population B
\item[\code{Nsel}] Loci under selection. NA under neutral selection
\item[\code{S}] Selection Coefficient
\end{ldescription}
\end{Value}
%
\begin{Author}\relax
F. Balao;
Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Wright, S. 1931. Evolution in Mendelian Populations. \emph{Genetics} \bold{16}: 97â159. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{hybridize}{hybridize}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=100, Na=30, Nb=30, Nf1=30, type="selection", Nsel=25, hybrid="F1")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sim2arlequin}{Convert a hybridsim object to a Arlequin input file}{sim2arlequin}
\keyword{hybridization}{sim2arlequin}
\keyword{simulation}{sim2arlequin}
%
\begin{Description}\relax
The function sim2arlequin converts a hybridsim object into a Arlequin input file. 

\end{Description}
%
\begin{Usage}
\begin{verbatim}
sim2arlequin(x,filename)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
a hybridsim object

\item[\code{filename}] 
a character string indicating the name of the output file
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Arlequin input file
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Arlequin
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{sim2bayescan}{sim2bayescan}}
\code{\LinkA{sim2introgress}{sim2introgress}}
\code{\LinkA{sim2newhybrids}{sim2newhybrids}}
\code{\LinkA{sim2popgene}{sim2popgene}}
\code{\LinkA{sim2structure}{sim2structure}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
F1hybrids<-hybridsim(Nmarker=100,Na=100,Nb=100,Nf1=30, type="neutral", hybrid="F1")

sim2arlequin(F1hybrids,filename="F1hybrids_Arlequin.txt")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sim2bayescan}{Convert a hybridsim object to a Bayescan input file}{sim2bayescan}
\keyword{hybridization}{sim2bayescan}
\keyword{simulation}{sim2bayescan}
%
\begin{Description}\relax
The function sim2bayescan converts a hybridsim object into a Bayescan (Foll \& Gaggiotti 2008) input file. 

\end{Description}
%
\begin{Usage}
\begin{verbatim}
sim2bayescan(x,filename)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
a hybridsim object. Only with F1 hybrids

\item[\code{filename}] 
a character string indicating the name of the output file
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Bayescan input file
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Foll, M., and O. Gaggiotti. 2008. A genome-scan method to identify selected loci appropriate for both dominant and codominant markers: a Bayesian perspective. \emph{Genetics} \bold{180}: 977â93. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{sim2arlequin}{sim2arlequin}}
\code{\LinkA{sim2introgress}{sim2introgress}}
\code{\LinkA{sim2newhybrids}{sim2newhybrids}}
\code{\LinkA{sim2popgene}{sim2popgene}}
\code{\LinkA{sim2structure}{sim2structure}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
F1hybrids<-hybridsim(Nmarker=100,Na=100,Nb=100,Nf1=30, type="selection", S=5, Nsel=25, hybrid="F1")

sim2bayescan(F1hybrids,filename="F1hybrids_Bayescan.txt")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sim2introgress}{Convert a hybridsim object to a introgress input file}{sim2introgress}
\keyword{hybridization}{sim2introgress}
\keyword{simulation}{sim2introgress}
%
\begin{Description}\relax
The function sim2introgress converts a hybridsim object into a introgress  input file. It is a wrapper to the function \code{\LinkA{prepare.data}{prepare.data}} of the package \pkg{introgress}

\end{Description}
%
\begin{Usage}
\begin{verbatim}
sim2introgress(x)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
a hybridsim object

\end{ldescription}
\end{Arguments}
%
\begin{Value}
introgress input file
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Gompert, Z., and C.A. Buerkle. 2010. introgress: a software package for mapping components of isolation in hybrids. \emph{Molecular Ecology Resources} \bold{10}: 378â384.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{prepare.data}{prepare.data}}
\code{\LinkA{sim2arlequin}{sim2arlequin}}
\code{\LinkA{sim2bayescan}{sim2bayescan}}
\code{\LinkA{sim2newhybrids}{sim2newhybrids}}
\code{\LinkA{sim2popgene}{sim2popgene}}
\code{\LinkA{sim2structure}{sim2structure}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=100,Na=100,Nb=100,Nf1=30, type="selection", hybrid="all")

hybrids2<-sim2introgress(hybrids)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sim2newhybrids}{Convert a aflpsim object to a NewHybrids input file}{sim2newhybrids}
\keyword{hybridization}{sim2newhybrids}
\keyword{simulation}{sim2newhybrids}
%
\begin{Description}\relax
The function sim2newhybrids converts a AFLPsim object into a NewHybrids input file. 

\end{Description}
%
\begin{Usage}
\begin{verbatim}
sim2newhybrids(x,filename)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
a aflpsim object

\item[\code{filename}] 
a character string indicating the name of the output file
\end{ldescription}
\end{Arguments}
%
\begin{Value}
NewHybrids input file
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Anderson, E.C. 2008. Bayesian inference of species hybrids using multilocus dominant genetic markers. \emph{Philosophical transactions of the Royal Society of London. Series B, Biological Sciences} \bold{363}: 2841â50. 

Anderson, E.C., and E.A. Thompson. 2002. A model-based method for identifying species hybrids using multilocus genetic data. \emph{Genetics} \bold{160}: 1217â1229.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{sim2arlequin}{sim2arlequin}}
\code{\LinkA{sim2bayescan}{sim2bayescan}}
\code{\LinkA{sim2introgress}{sim2introgress}}
\code{\LinkA{sim2popgene}{sim2popgene}}
\code{\LinkA{sim2structure}{sim2structure}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
hybrids<-hybridsim(Nmarker=500,Na=100,Nb=100,Nf1=30,Nbxa=30,Nbxb=30, Nf2=30, type="neutral", hybrid="all")

sim2newhybrids(hybrids,filename="F1newhybrids.txt")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sim2popgene}{Convert a hybridsim object to a PopGene input file}{sim2popgene}
\keyword{hybridization}{sim2popgene}
\keyword{simulation}{sim2popgene}
%
\begin{Description}\relax
The function sim2newhybrids converts a hybridsim object into a PopGene input file. 

\end{Description}
%
\begin{Usage}
\begin{verbatim}
sim2popgene(x,filename)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
a hybridsim object

\item[\code{filename}] 
a character string indicating the name of the output file
\end{ldescription}
\end{Arguments}
%
\begin{Value}
PopGene input file
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Yeh FC, Yang RC, Boyle TBJ, Ye ZH, Mao JX (1997) popgene, the User-Friendly Sharewarefor Population Genetic Analysis. Molecular Biology and Biotechnology Centre, University of Alberta, Canada (program available from: \url{http://www.ualberta.ca/~fyeh/}).
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{sim2arlequin}{sim2arlequin}}
\code{\LinkA{sim2bayescan}{sim2bayescan}}
\code{\LinkA{sim2introgress}{sim2introgress}}
\code{\LinkA{sim2newhybrids}{sim2newhybrids}}
\code{\LinkA{sim2structure}{sim2structure}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
F1hybrids<-hybridsim(Nmarker=100,Na=100,Nb=100,Nf1=30, type="neutral", hybrid="F1")

sim2popgene(F1hybrids,filename="F1hybrids_Popgene.txt")
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sim2structure}{Convert a hybridsim object to a Structure input file}{sim2structure}
\keyword{hybridization}{sim2structure}
\keyword{simulation}{sim2structure}
%
\begin{Description}\relax
The function sim2newhybrids converts a hybridsim object into a Structure 2.3 input file. 

\end{Description}
%
\begin{Usage}
\begin{verbatim}
sim2structure(x,filename)

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{x}] 
a hybridsim object

\item[\code{filename}] 
a character string indicating the name of the output file
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Structure input file
\end{Value}
%
\begin{Author}\relax
F. Balao \email{fbalao@us.es}, Garcia-CastaÃ±o J.L.
\end{Author}
%
\begin{References}\relax
Falush, D., M. Stephens, and J.K. Pritchard. 2007. Inference of population structure using multilocus genotype data: dominant markers and null alleles.\emph{Molecular Ecology Notes} \bold{7}: 574â578. 

Pritchard, J.K., M. Stephens, and P. Donnelly. 2000. Inference of population structure using multilocus genotype data. \emph{Genetics} \bold{155}: 945â959.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{sim2arlequin}{sim2arlequin}}
\code{\LinkA{sim2bayescan}{sim2bayescan}}
\code{\LinkA{sim2introgress}{sim2introgress}}
\code{\LinkA{sim2newhybrids}{sim2newhybrids}}
\code{\LinkA{sim2popgene}{sim2popgene}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
F1hybrids<-hybridsim(Nmarker=100,Na=100,Nb=100,Nf1=30, type="neutral", hybrid="F1")

sim2structure(F1hybrids,filename="F1hybrids_Structure.txt")
\end{ExampleCode}
\end{Examples}
\printindex{}
\end{document}
